ggshield/sca/output/schemas.py

```diff
@@ -0,0 +1,78 @@
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Literal, Optional, cast
+
+import marshmallow_dataclass
+from pygitguardian.models import Base, BaseSchema, FromDictMixin
+
+
+@dataclass
+class SCAVulnerability(Base, FromDictMixin):
+    severity: str
+    summary: str
+    cve_ids: list[str] = field(default_factory=list)
+    created_at: Optional[datetime] = None
+    fixed_version: Optional[str] = None
+
+
+SCAVulnerability.SCHEMA = cast(
+    BaseSchema,
+    marshmallow_dataclass.class_schema(SCAVulnerability, base_schema=BaseSchema)(),
+)
+
+SCADependencyType = Literal["direct", "transitive"]
```

Comment:
Would be nice if we could use an enum here, but maybe that's annoying to deserialize? 🤔

Reply:
Looks like we can make it work easily by doing `class SCADependencyType(str, Enum):` , [see here](https://stackoverflow.com/questions/24481852/serialising-an-enum-member-to-json), but serialization will directly transform it into a `str`.

Reply:
Based on your other answer, am I correct when I say these models are going to be used to parse Mescal responses?

If this is correct then what matters here is *de*-serialization. According to the stackoverflow answer you posted, using an enum would not help as one would need to manually turn the string into the enum (Unless Marshmallow can help here, but I doubt it). So `SCADependencyType` can stay the way it is, except... these models should not be in an `sca.output` package. If you look at `secret.output.schema`, it only contains models used to implement the json output. These would another case of models which start in ggshield and finish in py-gitguardian.

Reply:
Yes, sounds good.